### PR TITLE
Create Feature/sprint ghcn for 6 land converters during the ioda-converter code spring

### LINKED ISSUE #842: Upgrade converter for land/ghcn, imsfv3, owp, smap, smap9km, and smos to follow the JEDI Data Conventions
Update the converter identified in the title (hereafter referred to as the **target converter**) to follow the JEDI Data Conventions.

- Update variable names
- Ensure variable units match the units in the conventions table
- Update the global attributes dictionary as needed
- Run the converter test to make sure it runs (but will fail because of unupdated test reference)
- Run ```ioda-validate.x``` on your output file and verify that there are no errors or warnings
- Update test reference file
- Run unit tests making use of the target converter and verify that they pass
- Make sure you verify the results by examining the file using ```ncdump -t```, ```h5dump```, ```HDFview```, or a similar application.

## ZenHub organization

Please make sure that you have properly set:
- Labels: OBS3
- Estimate: An estimate of how long this issue will take to resolve. 1 = half a day. 2 = full day. 3 = two days, 5 = three days.
- Epics: [Sprint: Implement the JEDI Data Conventions in the Converters](https://github.com/JCSDA-internal/ioda-converters/issues/813)
- Assignees: Anyone **working** on this issue. Anyone who is not working directly on the issue, but
  who should receive notifications, should be listed below.
- Reviewers

## Notifications

Fill in a list of GitHub handles here. People here are not *assigned*, but instead receive notifications
about this issue's progress.

Notifications should always go to: @rhoneyager.

Other people:

### PR DESCRIPTION
**Pull request by YoulongXia-NOAA**
## Description

*(Instructions: replace text in this and all sections with your own text)*

-[] Following ioda-converter code sprint instructions, modifying and updating six ioda-converters for land:  **imsfv3_scf2ioda.py,    smap_ssm2ioda.py, ghcn_snod2ioda.py,  owp_snow_obs.py, smos_ssm2ioda.py,  smap9km_ssm2ioda.py**
- [] Removing afwa_snod2ioda.py from land

### Issue(s) addressed

Link the issues to be closed with this PR
- fixes #842 

If this does not close an existing issue, then [be sure to add a Pipeline, Label, Estimate, Assignees, and Epic](https://jointcenterforsatellitedataassimilation-jedi-docs.readthedocs-hosted.com/en/latest/inside/practices/issues.html)

## Acceptance Criteria (Definition of Done)

ctest passed and converters are merged into develop branch

## Dependencies

No

## Impact

Please list the other repositories (if any) that this PR will require changes in (example below)

- [ ] ufo

### CODE REVIEW

**Comment by YoulongXia-NOAA** [comment]
@rhoneyager @srherbener, now my converters have two problems: (a) I can set up for platform and sensor as a string rather than integer, and (b) for  smap9km_ssm2ioda.py, there are warning messages: 
 Variable MetaData/vegetationOpacity
 Warning: Variable 'MetaData/vegetationOpacity' is not listed in the YAML conventions file.
 Variable MetaData/surfaceFlag
 Warning: Variable 'MetaData/surfaceFlag' is not listed in the YAML conventions file.
Otherwise, all six converters are there.

**Comment by YoulongXia-NOAA** [comment]
Okay, I can use
 :platform = 789 ;
:sensor = 432 ; as an example. I will modify other converters related to satellite ID and sensor ID in the same way.

File: src/land/ghcn_snod2ioda.py
Comment (CoryMartin-NOAA) [inline src/land/ghcn_snod2ioda.py:181]: I think these are now supposed to be `dateTime`. May require my PR to be merged in before it will fully work.

File: src/land/imsfv3_scf2ioda.py
Comment (CoryMartin-NOAA) [inline src/land/imsfv3_scf2ioda.py:135]: `dateTime` again

File: src/land/owp_snow_obs.py
Comment (CoryMartin-NOAA) [inline src/land/owp_snow_obs.py:35]: Presumably `swe` also needs updated?

File: src/land/owp_snow_obs.py
Comment (CoryMartin-NOAA) [inline src/land/owp_snow_obs.py:109]: Remove commented out line if no longer needed

File: src/land/owp_snow_obs.py
Comment (CoryMartin-NOAA) [inline src/land/owp_snow_obs.py:206]: `dateTime` again

Comment (CoryMartin-NOAA) [reply src/land/owp_snow_obs.py:206]: Also can you condense these two lines into one?
`dim_dict['Location'] = len(self.data[('dateTime', 'MetaData')])`

File: src/land/smap9km_ssm2ioda.py
Comment (CoryMartin-NOAA) [inline src/land/smap9km_ssm2ioda.py:127]: Remove these commented out lines.

File: src/land/smap9km_ssm2ioda.py
Comment (CoryMartin-NOAA) [inline src/land/smap9km_ssm2ioda.py:136]: `dateTime` again

File: src/land/smos_ssm2ioda.py
Comment (CoryMartin-NOAA) [inline src/land/smos_ssm2ioda.py:126]: `dateTime` again

File: src/land/ghcn_snod2ioda.py
Comment (YoulongXia-NOAA) [reply src/land/ghcn_snod2ioda.py:181]: Corrected.

File: src/land/imsfv3_scf2ioda.py
Comment (YoulongXia-NOAA) [reply src/land/imsfv3_scf2ioda.py:135]: Corrected.

File: src/land/owp_snow_obs.py
Comment (YoulongXia-NOAA) [reply src/land/owp_snow_obs.py:35]: swe -> snowWaterEquivalent, units is changed: mm -> kg m-2

File: src/land/owp_snow_obs.py
Comment (YoulongXia-NOAA) [reply src/land/owp_snow_obs.py:206]: Yes, corrected and condensed.

File: src/land/smap9km_ssm2ioda.py
Comment (YoulongXia-NOAA) [reply src/land/smap9km_ssm2ioda.py:127]: Removed

File: src/land/smap9km_ssm2ioda.py
Comment (YoulongXia-NOAA) [reply src/land/smap9km_ssm2ioda.py:136]: Corrected

File: src/land/smos_ssm2ioda.py
Comment (YoulongXia-NOAA) [reply src/land/smos_ssm2ioda.py:126]: corrected.

**Comment by YoulongXia-NOAA** [comment]
@CoryMartin-NOAA, I have addressed all issues you indicated except for datetime ->dateTime. This issue will be addressed soon when other works are ready for it.

**Review by gthompsnJCSDA — CHANGES_REQUESTED** [review]
Where Cory marked the dateTime variables, you said you fixed/corrected them, yet when I checked out the branch, they were not changed.  This probably means you did not yet do a git add/commit/push of your changes.  Please do that soon.

Also, I found a couple spelling mistakes with your spelling of ```satelliate``` that should be fixed to ```satellite```

Otherwise, if the ioda-validate.x application works after these changes, then I am good to approve the PR.

**Comment by YoulongXia-NOAA** [comment]
> Where Cory marked the dateTime variables, you said you fixed/corrected them, yet when I checked out the branch, they were not changed. This probably means you did not yet do a git add/commit/push of your changes. Please do that soon.
> 
> Also, I found a couple spelling mistakes with your spelling of `satelliate` that should be fixed to `satellite`
> 
> Otherwise, if the ioda-validate.x application works after these changes, then I am good to approve the PR.

@gthompsnJCSDA, I changed datetime into dateTime, do make -j8 and run ctest, it aborted. I talked Cory and will wait for his work to be merged and see if I can run ctest. Two spelling of `satelliate` was fixed with `satellite`.

**Comment by CoryMartin-NOAA** [comment]
@gthompsnJCSDA this is because @YoulongXia-NOAA is using strings, so the PR you just approved (and then the one in ioda) is needed before Youlong's converter can use `dateTime` properly.

**Comment by YoulongXia-NOAA** [comment]
line 78-80, platform and sensor need to put integer. An example for SMAP is below:
self.satellite = 789
self.sensor = 432
AttrData["platform"] = np.array([self.satellite], dtype=np.int32)
AttrData["sensor"] = np.array([self.sensor], dtype=np.int32)

**Comment by YoulongXia-NOAA** [comment]
> @gthompsnJCSDA this is because @YoulongXia-NOAA is using strings, so the PR you just approved (and then the one in ioda) is needed before Youlong's converter can use `dateTime` properly.

I think that now most of us are using strings. We need some instructions and guidance to do further modifications. Thank you very much for your efforts in helping us to move forward. @CoryMartin-NOAA @gthompsnJCSDA

File: src/land/smap9km_ssm2ioda.py
Comment (CoryMartin-NOAA) [inline src/land/smap9km_ssm2ioda.py:128]: Remove commented out line

File: src/land/imsfv3_scf2ioda.py
Comment (CoryMartin-NOAA) [inline src/land/imsfv3_scf2ioda.py:89]: Is there an integer we can use instead?

File: src/land/smap9km_ssm2ioda.py
Comment (YoulongXia-NOAA) [reply src/land/smap9km_ssm2ioda.py:128]: Removed

File: src/land/imsfv3_scf2ioda.py
Comment (YoulongXia-NOAA) [reply src/land/imsfv3_scf2ioda.py:89]: Removed

**Comment by srherbener** [comment]
@gthompsnJCSDA, @CoryMartin-NOAA have your concerns been addressed? If so, please approve to indicate that. Thanks!

**Review by srherbener — APPROVED** [review]
Looks good (given that @CoryMartin-NOAA and @gthompsnJCSDA concerns are addressed). Thanks!

**Comment by srherbener** [comment]
> Looks good (given that @CoryMartin-NOAA and @gthompsnJCSDA concerns are addressed). Thanks!

And the conflicts are fixed. Thanks!

**Review by CoryMartin-NOAA — APPROVED** [review]

**Comment by gthompsnJCSDA** [comment]
@YoulongXia-NOAA   Did you make a corresponding pull request in the ioda:feature/sprint-ioda-converters to add the ```vegetationOpacity``` variable yet?  Also, there is already a variable called ```earthSurfaceType``` in the ```ObsSpace.yaml``` that you can use for the variable you have called ```surfaceFlag```.

**Comment by YoulongXia-NOAA** [comment]
> @YoulongXia-NOAA Did you make a corresponding pull request in the ioda:feature/sprint-ioda-converters to add the `vegetationOpacity` variable yet? Also, there is already a variable called `earthSurfaceType` in the `ObsSpace.yaml` that you can use for the variable you have called `surfaceFlag`.

Not yet, will do on Monday. I also need to solve the conflict simultaneously. Will let you know when I am ready. Thank you.

**Comment by YoulongXia-NOAA** [comment]
> @YoulongXia-NOAA Did you make a corresponding pull request in the ioda:feature/sprint-ioda-converters to add the `vegetationOpacity` variable yet? Also, there is already a variable called `earthSurfaceType` in the `ObsSpace.yaml` that you can use for the variable you have called `surfaceFlag`.

Not yet, will do on Monday. I also need to solve the conflict simultaneously. Will let you know when I am ready. Thank you.

**Comment by YoulongXia-NOAA** [comment]
Yes, it is https://github.com/JCSDA-internal/ioda/pull/659 @gthompsnJCSDA

**Comment by rhoneyager** [comment]
The ioda PR is merged. @gthompsnJCSDA, @YoulongXia-NOAA and I will look at the branch conflicts later today.

**Comment by YoulongXia-NOAA** [comment]
@Ryan, ctest fails is due that Andy code still use datetime rather than dateTime. After I changed, ctest works.  I also fixed for station_id with stationIdentification and ioda-validation.x passed.  So we can cancel tomorrow meeting. Thank you and have nice evening.

**Pull request merged**
